kernel/indTyping.ml

```diff
@@ -66,7 +66,9 @@ let mind_check_names mie =
 
 type univ_info = { ind_squashed : bool; ind_has_relevant_arg : bool;
                    ind_min_univ : Universe.t option; (* Some for template *)
-                   ind_univ : Universe.t }
+                   ind_univ : Universe.t;
+                   missing : Universe.Set.t; (* missing u <= ind_univ constraints *)
+                 }
```

Comment:
This looks a bit strange to me to have a field that is non-empty in case of failure. What about using a sum type and some monadic-like glue to carry on the error instead?

Reply:
We want to keep going after the first failure to accumulate others.
eg if someone sends the kernel `Inductive foo : Set := bla : Type@{u} -> Type@{v} -> foo.` we want to get both `u+1` and `v+1`

Reply:
Sure, but IIUC you just have to tweak `check_univ_leq`, the remaining code looks parametric in this type.

Reply:
Feel free to try, personally I don't see the point.

Reply:
Clear invariants. For instance, you shouldn't have `ind_squashed` to be `true` and a non-empty set of missing universes.